### ISSUE TITLE
Subtle errors in doc for keyboard modifier behavior

### ISSUE THREAD

**Issue by pushfoo** [pyglet/pyglet#629]
## tl;dr
1. Contrary to the current wording of the [keyboard modifiers doc](https://github.com/pyglet/pyglet/blob/master/doc/programming_guide/keyboard.rst#modifiers), modifiers such as capslock can be active without being held down
2. We should find out if there is any platform-specific behavior around when those modifiers are set and unset

Versions affected appear to include at least the following:
1. 1.5.17
2. 2.0.dev18
3. The tip of `master` branch

### Actual Behavior
My system details:
```
OS: Debian 11
Architecture: x86_64
DE: Gnome 3.38 on Wayland
```
On my system, the following are true:
1. Capslock & numlock (and maybe scrolllock) modifiers can be active without their corresponding key held down
2. Inactive lock modifiers become active **before** their key release event
3. Active lock modifiers become inactive **after** their key release event

I am unsure if items 2 and 3 in the list above are the same on all operating systems. It seems fastest to verify through experimentation.

### Verifying the Behavior
Save and run the example below locally, following the included instructions:
```py
"""
tl;dr demonstrate how some modifiers can be active w/o being held

Instructions for use:
    1. Launch this example
    2. Click the pyglet window to begin capturing key events
    3. Begin observing output
    4. Repeat steps 5-9 for the following: capslock, numlock, scrolllock
    5. Press some ordinary keys
    6. Press chosen lock key
    7. Press more ordinary keys
    8. Press chosen lock key again
    9. Review output further if necessary
"""
import pyglet


LOCK_MODS = (
    ('numlock', pyglet.window.key.MOD_NUMLOCK),
    ('capslock', pyglet.window.key.MOD_CAPSLOCK),
    ('scrolllock', pyglet.window.key.MOD_SCROLLLOCK)
)


def unpack_modifiers(modifiers, masks=LOCK_MODS, pad='  ') -> str:
    return ''.join(
        [f"{pad}{name}: {modifiers & mask}\n" for name, mask in masks]
    )


def print_event(event_name: str, key, modifiers) -> None:
        print(f"{event_name} key {key}:\n{unpack_modifiers(modifiers)}")


if __name__ == "__main__":
    window = pyglet.window.Window()

    @window.event
    def on_draw():
        window.clear()
        
    @window.event
    def on_key_press(key, modifiers):
        """Called whenever a key is pressed. """
        print_event("PRESS", key, modifiers)        

    @window.event
    def on_key_release(key, modifiers):
        """Called when the user releases a key. """
        print_event("RELEASE", key, modifiers)

    pyglet.app.run()
```
### Suggested Approach
1. Change the affected page to refer to 'currently active' modifiers rather than 'held down', and include an explanation of active vs held down
3. Verify whether modifier timing behavior is consistent across platforms before documenting it

I'll try to make a PR for item 1 this weekend, but I'll need help with item 2 for macOS and Windows. In the meantime, I'll try to obtain a cheap full-size keyboard to verify that scrolllock behaves the same way as numlock and capslock on Linux.

**Comment by benmoran56** [comment]
Agreed, I think this is just poor wording.

**Comment by benmoran56** [comment]
I did that small change. Another reference to "held down" was left alone, as it was specifically referring to SHIFT keys. 
I'll close this for now, but it's certainly OK if someone has time to expand this section of the docs with more detail.
